video/profiles.go

```diff
@@ -85,11 +85,44 @@ var DefaultProfile720p = EncodedProfile{
 // DefaultTranscodeProfiles defines the default set of encoding profiles to use when none are specified
 var DefaultTranscodeProfiles = []EncodedProfile{DefaultProfile360p, DefaultProfile720p}
 
-func GetPlaybackProfiles(iv InputVideo) ([]EncodedProfile, error) {
-	video, err := iv.GetTrack(TrackTypeVideo)
+func SetTranscodeProfiles(inputVideoStats InputVideo, reqTranscodeProfiles []EncodedProfile) ([]EncodedProfile, error) {
+
+	var transcodeProfiles []EncodedProfile
+	videoTrack, err := inputVideoStats.GetTrack(TrackTypeVideo)
 	if err != nil {
 		return nil, fmt.Errorf("no video track found in input video: %w", err)
 	}
+	// If Profiles haven't been overridden, use the default set
+	//get
+	if len(reqTranscodeProfiles) == 0 {
+		transcodeProfiles, err = GetDefaultPlaybackProfiles(videoTrack)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default transcode profiles: %w", err)
+		}
+		return transcodeProfiles, nil
+	} else if len(reqTranscodeProfiles) == 1 {
```

Comment:
Let's get a comment in here about why 1 profile is a special case

Reply:
Done